typescript/packages/subsurface-viewer/src/layers/map/privateMapLayer.ts

```diff
@@ -272,9 +272,15 @@ export default class PrivateMapLayer extends Layer<PrivateMapLayerProps> {
         const vertexIndex = 256 * 256 * r + 256 * g + b;
 
         if (typeof info.coordinate?.[2] !== "undefined") {
-            const depth = this.props.ZIncreasingDownwards
-                ? -info.coordinate[2]
-                : info.coordinate[2];
+            const zScale = this.props.modelMatrix
+                ? this.props.modelMatrix[10]
+                : 1;
+
+            const depth =
+                (this.props.ZIncreasingDownwards
+                    ? -info.coordinate[2]
+                    : info.coordinate[2]) / zScale;
```

Comment:
Can zScale be 0?

Reply:
No, but i can be arbitrarely small if one keeps on scaling down,

Reply:
I added a lower bound to it just to make sure.